code/api/utility.py

```diff
@@ -12,11 +13,55 @@ def getPaginated(data, page, size):
     except:
         return []
     
-def makePost(data):
+def makeLocalPost(data, author_id, post_id=None):
     """ 
-    Creates an InboxPost given json data and adds it to the given author_id's inbox
+    Creates a LocalPost given json data
     """
+    # save the received post as a LocalPost
+    received_post = LocalPost(
+            author_id=author_id,
+            title=data["title"],
+            description=data["description"],
+            content_type=data["contentType"],
+            content=data["content"].encode('utf-8'),
+            visibility=LocalPost.Visibility.get_visibility_choice(data["visibility"]),
+            unlisted=data["unlisted"],
+    )
+
+    received_post.save()
+    # set post origin and source to itself for a new post
+    received_post.origin = received_post.get_id()
+    received_post.source = received_post.get_id()
+    if post_id:
+        received_post.id = post_id
+    received_post.save()
```

Comment:
.save() called twice. lines 31 and 37

Reply:
Thanks, also removed a duplicate save in socialDistribution/views.py/posts